### ISSUE TITLE
Dependencies useless on running single test

### ISSUE THREAD

**Issue by marc-mabe** [Codeception/Codeception#4528]
#### What are you trying to achieve?

We have a bug test suite which takes a lot of time to finish.
There are some tests that depend on each other (e.g. a test for creating an entity requires another test to register a user and verify a new user)

As the hole test suite takes a lot of time I want to run a specific test only (e.g. creating an entity) but everything I tried results in failure as the depend tests don't get executed.

In general Codeception is configured to run everything on `./vendor/bin/codecept run`.
So to run only specific `SpotStatsBuildCest` which depends on others I tried the following:

Try 1:
`./vendor/bin/codecept run SpotStatsBuildCest`
```
Suite 'SpotStatsBuildCest' could not be found
```

Try 2:
`./vendor/bin/codecept run --fail-fast ./test/integration/cest/SpotStatsBuildCest.php`
```
Integration Tests (3) --------------------
S SpotStatsBuildCest: Run peak calculation job 
S SpotStatsBuildCest: Send peak success calculation notification mail 
S SpotStatsBuildCest: Send peak failed calculation notification mail 
---------------------------------------------------------------------
```

Try 3:
Added a `@group SpotStatsBuildCest` to the class and 
`./vendor/bin/codecept run --fail-fast --group SpotStatsBuildCest`
```
[Groups] SpotStatsBuildCest

Integration Tests (3) --------------------
S SpotStatsBuildCest: Run peak calculation job 
S SpotStatsBuildCest: Send peak success calculation notification mail 
S SpotStatsBuildCest: Send peak failed calculation notification mail 
---------------------------------------------------------------------
```

How can I run only this specific cest (+ dependencies)?

**Comment by DreadCube** [comment]
Please could we add something like that? :-) My testing crew would love that feature...

**Comment by Naktibalda** [comment]
Yes, this would be a really nice feature to have.

I will try to implement this, but I don't know when I will have time for it - it could take a few months.

**Comment by DanielNoga** [comment]
Neither beforeSuite() work when running single test

**Comment by Patabugen** [comment]
I'm having issues with this too - in my case I'm running the failed tests, but they're being skipped because the `testLogin` upon which they depend hasn't been run.

I would like for CodeCeption to either run the dependencies, even if not specified - or in the case of running failed tests (which I realise is hard to differentiate from any other group) to assume dependencies which are not included in the run have passed.

**Comment by vlad88sv** [comment]
Maybe adding a `--force` parameter that ignores dependencies could be an option too!

**Comment by marc-mabe** [comment]
Ignoring dependencies doesn't help in any way. There is a reason why a test has it and it's a requirement for this test. If your test does not have such a requirement you better don't mark it with dependent tests

**Comment by vlad88sv** [comment]
So I have a test that adds X component to a system and tests if it was saved, then I have another test that checks that configures X and stores it, and another one that checks that settings were properly stored.

Thing is that I might be back and forth doing changes on the 3rd test but I can't run it alone because it depends on #1 and #2 on a full test, but during development I would really like that dependencies were ignored.

**Comment by Patabugen** [comment]
Here's my use case: my tests are never dependent on another test running. I've got bootstrap code and factories for that.

I mostly use test dependencies for saying "This depends on LoginCest" because if LoginCest failed - there's no point running the rest of the tests.

Another option (in addition to my previosu suggestion about runnign the dependnecies even if they're not selected) would be to have a new attribute "@runFirst LoginCest" so it'll always make sure that test has been run before running the current test.

**Comment by cjshowalter** [comment]
@marc-mabe I think your try 1 fails since you have to say the filename "codecept run SpotStatsBuildCest.php" instead. For your Try 3, did you try adding all the tests it depends on to that same group as well? I tried that with a really small test and it worked.
Even if that works, having @depends guarantee that the other tests will run (or something of the sort) would be really nice...

**Comment by diogoca** [comment]
any news about that? I got stuck on same...

**Comment by rogerpro** [comment]
I ask the same, any update? IMHO this is a really nice to have, and a must for those developing with TDD. How do you rerun a specific test or class whilst developing? Waiting for 2 minutes for the whole test suite is no needed when you are focused on the response of a specific request (that may have dependencies when running a full test in a clean environment).

Personally, I just add a `_` to the annotation so Codeception ignores `@depends_` and there's no need to rollback huge amount of annotations after the feature is done, but I guess that you agree that this is ugly and not a pro way.

I would love an `--ignore-dependencies` that would simply ignore the `@depends` annotations of the tests ran, being them called either by group, class or single test.

@Naktibalda, is there a way that we can help you with that? I haven't dive into the code but I'm open for it if it helps.

Thank you very much and let's make Codeception the perfect tool :-)

**Comment by rogerpro** [comment]
> Ignoring dependencies doesn't help in any way. There is a reason why a test has it and it's a requirement for this test. If your test does not have such a requirement you better don't mark it with dependent tests

I agree with that when the tests run in a pipeline or before pushing, for instance.

But, when developing in TDD, we first write a very specific test (typically, for a single endpoint) and we expect different cases of responses. At that moment, we use to have the needed data available in the database, maybe testing data. And we first focus on this endpoint, so we don't need to run dozens of tests on every small change that we do.

When we are done with this particular test, then of course will run again all the tests with a clean database and taking into account the dependencies. That's why we need 2 modes of running tests without rewriting them: with and without dependencies.
